tests/reporting/test_reporting.py

```diff
@@ -1,108 +1,442 @@
+    reputationToken = sessionFixture.applySignature('ReputationToken', branch.getReputationToken())
     legacyRepContract.approve(reputationToken.address, 11 * 10**6 * 10**18)
     reputationToken.migrateFromLegacyRepContract()
 
-    # give some REP to someone else to make things interesting
-    reputationToken.transfer(tester.a1, 4 * 10**6 * 10**18)
-    reputationToken.transfer(tester.a2, 1 * 10**6 * 10**18)
+    # Give some REP to testers to make things interesting
+    for testAccount in [tester.a1, tester.a2, tester.a3]:
+        reputationToken.transfer(testAccount, 1 * 10**6 * 10**18)
 
-    # buy registration tokens
-    reportingWindow = contractsFixture.applySignature('ReportingWindow', market.getReportingWindow())
-    firstRegistrationToken = contractsFixture.applySignature('RegistrationToken', reportingWindow.getRegistrationToken())
+    reportingWindow = sessionFixture.applySignature('ReportingWindow', market.getReportingWindow())
+    firstRegistrationToken = sessionFixture.applySignature('RegistrationToken', reportingWindow.getRegistrationToken())
+    reportingTokenNo = sessionFixture.getReportingToken(market, [2,0])
+    reportingTokenYes = sessionFixture.getReportingToken(market, [0,2])
+
+    # Tester 0, 1, 2, and 3 will buy registration tokens so they can report on the market
     firstRegistrationToken.register(sender=tester.k0)
     assert firstRegistrationToken.balanceOf(tester.a0) == 1
-    assert reputationToken.balanceOf(tester.a0) == 6 * 10**6 * 10**18 - 10**18
-    firstRegistrationToken.register(sender=tester.k1)
-    assert firstRegistrationToken.balanceOf(tester.a1) == 1
-    assert reputationToken.balanceOf(tester.a1) == 4 * 10**6 * 10**18 - 10**18
-
-    # fast forward to one second after the next reporting window
-    reportingWindow = contractsFixture.applySignature('ReportingWindow', branch.getNextReportingWindow())
-    contractsFixture.chain.head_state.timestamp = reportingWindow.getStartTime() + 1
-
-    # both reporters report on the outcome
-    reportingTokenNo = contractsFixture.getReportingToken(market, [2,0])
-    reportingTokenYes = contractsFixture.getReportingToken(market, [0,2])
+    # Testers will have their previous REP balance less the registration token bond of 1 REP
+    assert reputationToken.balanceOf(tester.a0) == 8 * 10**6 * 10**18 - 10**18
+
+    for (key, address) in [(tester.k1, tester.a1), (tester.k2, tester.a2), (tester.k3, tester.a3)]:
+        firstRegistrationToken.register(sender=key)
+        assert firstRegistrationToken.balanceOf(address) == 1
+        assert reputationToken.balanceOf(address) == 1 * 10**6 * 10**18 - 10**18
+
+    sessionFixture.reputationToken = reputationToken
+    sessionFixture.reportingWindow = reportingWindow
+    sessionFixture.firstRegistrationToken = firstRegistrationToken
+    sessionFixture.reportingTokenNo = reportingTokenNo
+    sessionFixture.reportingTokenYes = reportingTokenYes
```

Comment:
This will add these variables to the `sessionFixture` for everyone.  This isn't directly a problem, but it may result in unexpected behavior where if the tests are run before a test that uses `reportingSnapshot` they will behave differently from tests that are run after a test that references `reportingSnapshot`.

I recommend creating a separate class to hold these objects rather than applying them to `sessionFixture`.  This way you only get access to these variables if you depend on `reportingSnapshot` and you can't get access to them by only depending on `sessionFixture` and happening to have some other test run prior to yours that depends on `reportingSnapshopt`.